lib/closure_tree/has_closure_tree.rb

```diff
@@ -1,38 +1,38 @@
 module ClosureTree
   module HasClosureTree
     def has_closure_tree(options = {})
-      options.assert_valid_keys(
-        :parent_column_name,
-        :dependent,
-        :hierarchy_class_name,
-        :hierarchy_table_name,
-        :name_column,
-        :order,
-        :dont_order_roots,
-        :numeric_order,
-        :touch,
-        :with_advisory_lock
-      )
+      connection_pool.with_connection {
+        options.assert_valid_keys(
```

Comment:
Maybe the check could happen before (or outside) the `connection_pool.with_connection` block to avoid useless connections?